python/cudf/cudf/core/column/string.py

```diff
@@ -879,10 +879,11 @@ def split(self, pat=None, n=-1, expand=True, **kwargs):
         from cudf._libxx.scalar import Scalar
 
         result_table = cpp_split(self._column, Scalar(pat, "str"), n)
-
         if len(result_table._data) == 1:
             if result_table._data[0].null_count == len(self._parent):
                 result_table = []
+            elif self._parent.null_count == len(self._parent):
```

Comment:
`self._parent` can be `None`:
```suggestion
            elif self._column.null_count == len(self._column):
```